### ISSUE TITLE
Nullability is ignored for `foreach` iterator variable

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#26631]
No warnings reported for `x.ToString()` or `y.ToString()`:
```
using System.Collections.Generic;
class C
{
    static void F(IEnumerable<object?> cx, object?[] cy)
    {
        foreach (var x in cx)
        {
            x.ToString(); // may be null
        }
        foreach (object? y in cy)
        {
            y.ToString(); // may be null
        }
    }
}
```
